app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
+  def self.generate_authorization_code(user:, client_id:, nonce:)
+    identity = user.identities.create!(
+      service_provider: client_id,
+      session_uuid: SecureRandom.uuid,
+      last_authenticated_at: Time.current,
+      nonce: nonce
+    )
+
+    identity.session_uuid
+  end
```

Comment:
Yeah I wasn't sure how to reuse the code from IdentityLinker --- the main difference being that we want to be able to set the `nonce` here. Should I just update IdentityLinker to take an optional `nonce`?